vm/interpreter/src/circ_supply.rs

```diff
@@ -0,0 +1,254 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use actor::*;
+use address::Address;
+use blocks::Tipset;
+use chain::*;
+use cid::Cid;
+use clock::ChainEpoch;
+use fil_types::{
+    FILECOIN_PRECISION, FIL_RESERVED, UPGRADE_ACTORS_V2_HEIGHT, UPGRADE_IGNITION_HEIGHT,
+    UPGRADE_LIFTOFF_HEIGHT,
+};
+use ipld_blockstore::BlockStore;
+use num_bigint::BigInt;
+use state_tree::StateTree;
+use std::collections::HashMap;
+use std::error::Error as StdError;
+
+#[derive(Default)]
+pub struct GenesisInfo {
+    genesis_msigs: Vec<multisig::State>,
+    // info about the Accounts in the genesis state
+    genesis_actors: Vec<GenesisActor>,
+    genesis_pledge: TokenAmount,
+    genesis_market_funds: TokenAmount,
+}
+
+pub struct GenesisActor {
+    addr: Address,
+    init_bal: TokenAmount,
+}
+
+fn get_actor_state<DB: BlockStore>(
+    state_tree: &StateTree<DB>,
+    addr: &Address,
+) -> Result<ActorState, ActorError> {
+    state_tree
+        .get_actor(&addr)
+        .map_err(|e| {
+            e.downcast_default(
+                ExitCode::ErrIllegalState,
+                "failed to get reward actor for cumputing total supply",
+            )
+        })?
+        .ok_or_else(|| actor_error!(ErrIllegalState; "Actor address ({}) does not exist", addr))
+}
+
+pub fn get_fil_vested<DB: BlockStore>(
+    pre_ignition: &GenesisInfo,
+    post_ignition: &GenesisInfo,
+    height: ChainEpoch,
+    state_tree: &StateTree<DB>,
+) -> Result<TokenAmount, ActorError> {
```

Comment:
This can just be a boxed std Error or even string error, there is no need for ActorError outside of the VM, and when it's called within, everything is overridden to `ErrIllegalState`